src/test/java/com/example/demo/controller/BookControllerTest.java

```diff
@@ -0,0 +1,214 @@
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(
+                    "database/books/remove-all-books.sql"));
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "user", roles = {"USER"})
+    @DisplayName("method returns all available books")
+    @Transactional
+    void getAll_ValidRequest_ShouldReturnAllAvailableBooks() throws Exception {
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(new BookDto().setId(1L)
+                                  .setTitle("Book Title1")
+                                  .setAuthor("Author Name1")
+                                  .setIsbn("1234567890123")
+                                  .setPrice(BigDecimal.valueOf(23.99))
+                                  .setDescription("Book description1")
+                                  .setCoverImage("image_url1")
+                                  .setCategories(new HashSet<>()));
+        expected.add(new BookDto().setId(2L)
+                                  .setTitle("Book Title2")
+                                  .setAuthor("Author Name2")
+                                  .setIsbn("12345678901232")
+                                  .setPrice(BigDecimal.valueOf(45.99))
+                                  .setDescription("Book description2")
+                                  .setCoverImage("image_url2")
+                                  .setCategories(new HashSet<>()));
+
+        MvcResult result = mockMvc.perform(get("/books"))
+                                  .andExpect(status().isOk())
+                                  .andReturn();
+        BookDto[] actual = objectMapper.readValue(
+            result.getResponse()
+                  .getContentAsString(), BookDto[].class);
+        Assertions.assertEquals(expected.size(), actual.length);
+        Assertions.assertEquals(expected, Arrays.stream(actual)
+                                                .toList());
+    }
+
+    @Test
+    @WithMockUser(username = "user", roles = {"USER"})
+    @DisplayName("method returns book by id")
+    @Transactional
+    void getBookById_ValidId_ShouldReturnBookDto() throws Exception {
+        BookDto expected = new BookDto().setId(2L)
+                                        .setTitle("Book Title2")
+                                        .setAuthor("Author Name2")
+                                        .setIsbn("12345678901232")
+                                        .setPrice(BigDecimal.valueOf(45.99))
+                                        .setDescription("Book description2")
+                                        .setCoverImage("image_url2")
+                                        .setCategories(new HashSet<>());
```

Comment:
if you use same books lets create method that build this entity and properties better save to constants and reuse it